### PR TITLE
Feature/rspec (part1 + part2)

### PR DESCRIPTION
**Pull request by rodikno**
[RSpec Core, Part 1]
1. Create unit tests for couple your Trello models
2. Create appropriate files and folders structure needed to add an RSpec test solution to our current main project
3. Implement all your current test scenarios using PageObject gem
- use describes and its
- use before hooks to execute prerequisites

[RSpec Core, Part 2]
- change describes to contexts where it's needed
- use lets where you do some test related actions
- create helper methods where you need it and use them
- try to use subject and 'is_expected' expressions
- add appropriate tags to describe blocks
- add rake tasks to run rspec tests
- find appropriate cases and use shared_examples and shared_contexts
- create 3 more rake tasks which could run with different formatters

### CODE REVIEW

File: helpers/redmine_helper.rb [hunk 1]
```diff
@@ -37,4 +37,12 @@ def is_http_get_response_positive?(url)
     end
   end
 
+  def log_out
+    begin
+      @browser.find_element(css: '.logout').click
+    rescue Selenium::WebDriver::Error::NoSuchElementError
```
Comment (d-konovalov) [inline helpers/redmine_helper.rb:43]: if you rescue some error you should provide action you'd want your code to apply if this error appeared
empty rescue block it is very bad option

File: spec/redmine/redmine_issue_spec.rb [hunk 2]
```diff
@@ -0,0 +1,59 @@
+require './spec/spec_helper'
+
+describe '[Issue]', :redmine, :issue do
+  user = RedmineUser.new
+  project = RedmineProject.new
```
Comment (d-konovalov) [inline spec/redmine/redmine_issue_spec.rb:5]: these lines are kind of prerequisite for the tests
so they should be placed inside before or let block or even moved into shared_context blocks at all

**Review by d-konovalov — APPROVED** [review]

File: helpers/redmine_helper.rb (hunk 1 above)
Comment (rodikno) [reply helpers/redmine_helper.rb:43]: Hi @d-konovalov, thank you for you review)

This method could be called when user is not logged in in some cases so I've just added `return` here now to have some meaningful result action (previously the message `User is already logged out` was returned, but it was spoiling reports all the time)

File: spec/redmine/redmine_issue_spec.rb (hunk 2 above)
Comment (rodikno) [reply spec/redmine/redmine_issue_spec.rb:5]: I was trying to find a solution here, so the 'prerequisite' objects could be available in:
- relative and nested example groups and examples
- all relative and nested lets, befores and afters
- (!) as an argument for nested and relative `include_context` and `include_examples` (which is a main issue now)

And I can see two options:
1) I'm structuring specs wrong so it makes me start looking for a solution here
2) Or I can use instance variables in `before` hooks like

```Ruby
describe '[Issue]' do
  before :all do
    @user = RedmineUser.new
    @project = RedmineProject.new
  end
  context 'I am nested context' do
    include_examples 'Register as a', @user
    include_examples 'Create a new project', @project
    # nested contexts and examples which could use these variables unlimitly
  end
end
```
https://www.evernote.com/shard/s19/sh/9e13bd34-5db5-49b1-aa5c-03fc8310d88e/bb656f5793aee5591a36c87b5ff7ec8a

Hope you wouldn't mind if I come to your table tomorrow :)

**Pull request merged**